### DISCUSSION TITLE
Covering Bbox properties xmin/ymax/ymin/ymax  are of type array?

### DISCUSSION THREAD

**Discussion by bertt** [opengeospatial/geoparquet#238] [Ecosystem]
Hi, I was looking at the JSON Schema 1.1 (https://geoparquet.org/releases/v1.1.0/schema.json). 
I noticed the covering  Bbox properties xmin/ymax/ymin/ymax  are of type array:

```
                    "xmin": {
                      "type": "array",
                      "items": [
                        {
                          "type": "string",
                          "minLength": 1
                        },
                        {
                          "const": "xmin"
                        }
                      ],
                      "minItems": 2,
                      "maxItems": 2
                    },
```

Why it's not just a number?

**Comment by jwass** [comment]
Hi @bertt, the "covering" element refers to the column names that contain the bounding box for each row. Some more details are here: https://github.com/opengeospatial/geoparquet/blob/main/format-specs/geoparquet.md#covering.

The top-level `bbox` parameter (https://github.com/opengeospatial/geoparquet/blob/main/format-specs/geoparquet.md#bbox) is defined here https://github.com/opengeospatial/geoparquet/blob/main/format-specs/schema.json#L54. That value an array of numbers of the bounding box corners for the entire geoparquet file - which might be what you're looking for(?)

**Comment by bertt** [comment]
Ah ok thanks, so the covering bbox is per row... Maybe there is an example file with this covering bbox somewhere?

**Reply by jwass** [reply]
Overture Maps publishes GeoParquet files with the row-level bounding box. For example - picking a random file from the buildings theme in the most recent release: s3://overturemaps-us-west-2/release/2024-06-13-beta.1/theme=buildings/type=building/part-00212-6bb975da-d9d5-4823-b446-c0844c96c235-c000.zstd.parquet. We're not writing out the GeoParquet 1.1 metadata yet so the covering field is not populated in the parquet metadata - but the bbox column has that data for every row.

However recent versions of GDAL and ogr2ogr will write out that covering field if you want to see a file with it: `ogr2ogr -of Parquet dest.parquet src.geojson`

**Comment by bertt** [comment]
Sample for countries worldwide (generated by GDAL 3.10.0dev):

```
{
   "version":"1.1.0",
   "primary_column":"geometry",
   "columns":{
      "geometry":{
         "encoding":"WKB",
         "bbox":[
            -180.0,
            -90.0,
            180.0,
            90.0
         ],
         "covering":{
            "bbox":{
               "xmin":[
                  "geometry_bbox",
                  "xmin"
               ],
               "ymin":[
                  "geometry_bbox",
                  "ymin"
               ],
               "xmax":[
                  "geometry_bbox",
                  "xmax"
               ],
               "ymax":[
                  "geometry_bbox",
                  "ymax"
               ]
            }
         },
         "orientation":"counterclockwise",
         "geometry_types":[
            "MultiPolygon"
         ]
      }
   }
}
```

**Comment by bertt** [comment]
Addition: When opening the GeoParquet file in QGIS the extra columns geometry_bbox.xmin, geometry_bbox.xmax, geometry_bbox.ymin, geometry_bbox.ymax are there:

![image](https://github.com/opengeospatial/geoparquet/assets/538812/1dac1d60-6162-47fd-8ce1-b50d51c1d5e9)

**Reply by rouault** [reply]
> When opening the GeoParquet file in QGIS the extra columns geometry_bbox.xmin, geometry_bbox.xmax, geometry_bbox.ymin, geometry_bbox.ymax are there:

Probably because its GDAL version is < 3.9. GDAL >= 3.9 will hide the GeoParquet gometry_bbox columns.
